src/finch/tensor.py

```diff
@@ -38,105 +74,138 @@ def ndim(self) -> int:
         return jl.ndims(self._obj)
 
     @property
-    def shape(self) -> Tuple[int, ...]:
+    def shape(self) -> tuple[int, ...]:
         return jl.size(self._obj)
 
     @property
     def size(self) -> int:
         return np.prod(self.shape)
 
+    @property
+    def _is_dense(self) -> bool:
+        lvl = self._obj.body.lvl
+        for _ in self.shape:
+            if not jl.isa(self._obj, jl.Finch.Dense):
+                return False
+            lvl = lvl.lvl
+        return True
+
+    @property
+    def _order(self) -> tuple[int, ...]:
+        return jl.typeof(self._obj).parameters[1]
+
+    def get_order(self, zero_indexing=True) -> tuple[int, ...]:
+        order = self._order
+        if zero_indexing:
+            order = tuple(i - 1 for i in order)
+        return order
+
+    def get_inv_order(self, zero_indexing=True) -> tuple[int, ...]:
+        inv_order = jl.invperm(self._order)
+        if zero_indexing:
+            inv_order = tuple(i - 1 for i in inv_order)
+        return inv_order
+
     def __pos__(self):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._obj), order=self.order)
```

Comment:
is self.order defined?

Reply:
Moreover, does this rely on the broadcast returning a tensor and not just a swizzle?